src/main/java/com/snowflake/kafka/connector/internal/ingestsdk/ClientManager.java

```diff
@@ -0,0 +1,122 @@
+    this();
+    this.streamingIngestClient = client;
+  }
+
+  /**
+   * Calls the ingest sdk to create the streaming client
+   *
+   * @param connectorConfig properties for the streaming client
+   * @param kcInstanceId identifier of the connector instance creating the client
+   */
+  public void createStreamingClient(Map<String, String> connectorConfig, String kcInstanceId) {
+    Map<String, String> streamingPropertiesMap =
+        StreamingUtils.convertConfigForStreamingClient(new HashMap<>(connectorConfig));
+    Properties streamingClientProps = new Properties();
+    streamingClientProps.putAll(streamingPropertiesMap);
+
+    String streamingIngestClientName = this.getStreamingIngestClientName(kcInstanceId);
+
+    try {
+      LOGGER.info("Creating Streaming Client. ClientName:{}", streamingIngestClientName);
+      this.streamingIngestClientCount++;
+      this.streamingIngestClient =
+          SnowflakeStreamingIngestClientFactory.builder(streamingIngestClientName)
+              .setProperties(streamingClientProps)
+              .build();
+    } catch (SFException ex) {
+      LOGGER.error(
+          "Exception creating streamingIngestClient with name:{}", streamingIngestClientName);
+      throw new ConnectException(ex);
+    }
+  }
+
+  /**
+   * Calls the ingest sdk to close the client sdk
+   *
+   * @return true if the client was successfully closed, false if not
+   */
+  public boolean closeStreamingClient() {
+    if (this.streamingIngestClient == null || this.streamingIngestClient.isClosed()) {
+      LOGGER.info("Streaming client is already closed or null");
+      return true;
+    }
+
+    String streamingIngestClientName = this.streamingIngestClient.getName();
+    LOGGER.info("Closing Streaming Client:{}", streamingIngestClientName);
+
+    try {
+      this.streamingIngestClient.close();
+      return true;
+    } catch (Exception e) {
+      String message =
+          e.getMessage() != null && !e.getMessage().isEmpty()
+              ? e.getMessage()
+              : "no error message provided";
+
+      String cause =
+          e.getCause() != null
+                  && e.getCause().getStackTrace() != null
+                  && !Arrays.toString(e.getCause().getStackTrace()).isEmpty()
+              ? Arrays.toString(e.getCause().getStackTrace())
+              : "no cause provided";
+
+      LOGGER.error("Failure closing Streaming client msg:{}, cause:{}", message, cause);
+      return false;
+    }
+  }
+
+  /**
+   * Gets the streaming client if it was created
+   *
+   * @return The streaming client, throws an exception if no client was initialized
+   */
+  public SnowflakeStreamingIngestClient getStreamingIngestClient() {
+    if (this.streamingIngestClient != null && !this.streamingIngestClient.isClosed()) {
+      return this.streamingIngestClient;
+    }
+
+    LOGGER.error("Streaming ingest client was null or closed. It must be initialized");
+    throw SnowflakeErrors.ERROR_3009.getException();
```

Comment:
So some methods seem like they throw an exception if the client is null or closed but some (like `close`) silently succeed. We might want to think about consolidating behaviors across all methods.

Reply:
This is true, looking at the KcStreamingIngestClient.java we basically just wrap the streaming ingest client. This way our behavior is the same as the streaming ingest client. We also do a null check at the end of the constructor to ensure theres no nullPointerExceptions. The only major change we make is to swallow the close exception because closing the client is best effort